src/firmware/guest/mod.rs

```diff
@@ -86,23 +86,62 @@ impl Firmware {
         &mut self,
         message_version: Option<u8>,
         report_request: SnpReportReq,
-    ) -> Result<(AttestationReport, CertTable), Indeterminate<Error>> {
+    ) -> Result<(AttestationReport, Vec<CertTableEntry>), Indeterminate<Error>> {
+        // Due to the complex buffer allocation, we will take the SnpReportReq
+        // provided by the caller, and create an extended report request object
+        // for them.
         let ext_report_request: SnpExtReportReq = SnpExtReportReq::new(report_request);
 
-        let ext_report_response: SnpReportRsp = SnpReportRsp::default();
-        let mut cert_table: CertTable = Default::default();
-
-        SNP_GET_EXT_REPORT.ioctl(
-            &mut self.0,
-            &mut SnpGuestRequest::new(message_version, &ext_report_request, &ext_report_response),
-        )?;
+        // Create an object for the PSP to store the response content in.
+        let ext_report_response: SnpReportRsp = Default::default();
```

Comment:
Should this variable be `mut` because the ioctl will modify its content?

Reply:
I suppose it wouldn't hurt, but I'm not entirely sure it's needed. When we use this response to build an `SnpGuestRequest` [here](https://github.com/virtee/sev/pull/54/files#diff-f2a75c2a321a96e5bda859cce9503572e03a5068d455ee0c5a276dad6d949800R105), the request takes a pointer to this struct. The ioctl itself is inherently unsafe, as Rust can't guarantee any safety measures with what the kernel does with this data.